File: P1-Linguagem_de_Programação_I/Desafios dos Roteiros - C++/desafio-roteiro03-parte06.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){

    vector<string> nomes;
    vector<int> quantidades;
    vector<double> precos;

    int opcao = 0;

    while(opcao != 3){

        cout << "\n1 - Adicionar produto" << endl;
        cout << "2 - Listar produtos" << endl;
        cout << "3 - Sair" << endl;

        cout << "Escolha uma opcao: ";
        cin >> opcao;

        if(opcao == 1){

            string nome;
            int qtd;
            double preco;

            cout << "Nome do produto: ";
            cin >> nome;

            cout << "Quantidade: ";
            cin >> qtd;

            if(qtd < 0){
                cout << "Erro: quantidade nao pode ser negativa." << endl;
                continue;
            }

            cout << "Preco: ";
            cin >> preco;

            nomes.push_back(nome);
            quantidades.push_back(qtd);
            precos.push_back(preco);

        }

        else if(opcao == 2){

            for(int i = 0; i < nomes.size(); i++){
                cout << "\nProduto: " << nomes[i] << endl;
                cout << "Quantidade: " << quantidades[i] << endl;
                cout << "Preco: " << precos[i] << endl;
            }

        }
    }

    cout << "Sistema encerrado." << endl;

    return 0;
}

File: P1-Linguagem_de_Programação_I/Desafios dos Roteiros - Python/desafio-roteiro03-parte06.py
produtos = []
quantidades = []
precos = []

opcao = 0

while opcao != 3:

    print("\n1 - Adicionar produto")
    print("2 - Listar produtos")
    print("3 - Sair")

    opcao = int(input("Escolha uma opção: "))

    if opcao == 1:
        nome = input("Nome do produto: ")
        qtd = int(input("Quantidade: "))

        if qtd < 0:
            print("Erro: quantidade não pode ser negativa.")
            continue

        preco = float(input("Preço: "))

        produtos.append(nome)
        quantidades.append(qtd)
        precos.append(preco)

    elif opcao == 2:

        for i in range(len(produtos)):
            print("\nProduto:", produtos[i])
            print("Quantidade:", quantidades[i])
            print("Preço:", precos[i])

print("Sistema encerrado.")